pkg/kallsyms/kallsyms.go

```diff
@@ -149,20 +154,34 @@ var (
 //
 //	kallsyms.SpecUpdateAddresses(spec, []string{"socket_file_ops"})
 func SpecUpdateAddresses(spec *ebpf.CollectionSpec, symbols []string) error {
-	kAllSymsFactory := NewKAllSyms
-	return specUpdateAddresses(kAllSymsFactory, spec, symbols)
+	if len(symbols) == 0 {
+		// Nothing to do
+		return nil
+	}
+
+	var err error
+	once.Do(func() {
+		kallsyms, err = NewKAllSyms()
+	})
+	if err != nil {
+		return err
```

Comment:
I don't think the `err != nil` test would capture errors correctly for subsequent calls to SpecUpdateAddresses.
See https://go.dev/play/p/2RRA9mrM8Fq

The `err` variable needs to be a global variable next to once line 31.

Reply:
Good catch! I fixed it.